### DISCUSSION TITLE
Debug System Perfomance

### DISCUSSION THREAD

**Discussion by MaansenV** [friflo/Friflo.Json.Fliox#47] [Q&A] [answered]
Hey, just started with this lib, and liking it :).

How would i track the Perfomance each System is taking? so i can track bottle necks

**Comment by friflo** [comment, marked as answer]
Hi, appreciate that you like it.

You just enable monitoring with [SystemGroup.SetMonitorPerf](https://github.com/friflo/Friflo.Engine-docs/blob/main/api/SystemGroup.SetMonitorPerf(bool).md)

```csharp
var root = new SystemRoot("Systems");
root.SetMonitorPerf(true);
```

Each system has a `Perf` property of type [SystemPerf](https://github.com/friflo/Friflo.Engine-docs/blob/main/api/SystemPerf.md).
It provides the time of the last execution and the sum of all execution times in milliseconds and system ticks.
